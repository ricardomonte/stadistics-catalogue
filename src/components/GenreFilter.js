import React from 'react';
import PropTypes from 'prop-types';
import GameStyle from '../styles/Filter.module.css';

const GenreFilter = ({ filter, onGameChange }) => {
  const genresArray = [
    'ALL',
    'Racing',
    'Fighting',
    'Shooter',
    'Puzzle',
    'Role-playing (RPG)',
    'Simulator',
    'Platform',
    'Adventure',
    'Quiz/Trivia',
    'Sport',
  ];

  return (
    <div className={GameStyle.filter}>
      <p className={GameStyle.filterText}>Filter:</p>
      <select
        id="genres"
        name="genres"
        value={filter}
        onChange={onGameChange}
        className={GameStyle.selection}
      >
        <option value="" disabled>
          Genres...
        </option>
        {genresArray.map((genre) => (
          <option key={genre} value={genre}>
            {genre}
          </option>
        ))}
      </select>
    </div>
  );
};

GenreFilter.propTypes = {
  filter: PropTypes.string.isRequired,
  onGameChange: PropTypes.func.isRequired,
};
export default GenreFilter;
