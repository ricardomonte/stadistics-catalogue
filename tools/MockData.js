const MockData = [
  {
    id: 1,
    name: 'test data 1',
    cover: '//mock.data.com/test.jpg',
    genre: 'Strategy',
    rating: 72,
  },
  {
    id: 2,
    name: 'test data 2',
    cover: '//mock.data.com/test2.jpg',
    genre: 'Sports',
    rating: 80,
  },
  {
    id: 3,
    name: 'test data 3',
    cover: '//mock.data.com/test3.jpg',
    genre: 'Sports',
    rating: 10,
  },
  {
    id: 4,
    name: 'test data 4',
    cover: '//mock.data.com/test4.jpg',
    genre: 'Fighting',
    rating: 95.67,
  },
  {
    id: 5,
    name: 'test data 5',
    cover: '//mock.data.com/test5.jpg',
    genre: 'Arcade',
    rating: 100,
  },
];

export default MockData;
