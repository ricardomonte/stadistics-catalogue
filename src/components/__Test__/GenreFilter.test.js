import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import MockData from '../../../tools/MockData';
import GenreFilter from '../GenreFilter';

test('Content from GenreFilter matches snapshot', () => {
  const content = render(
    <GenreFilter filter={MockData[0].genre} onGameChange={jest.fn()} />,
  );
  expect(content.container).toMatchSnapshot();
  expect(screen.getByText('Filter:')).toBeInTheDocument();
});
