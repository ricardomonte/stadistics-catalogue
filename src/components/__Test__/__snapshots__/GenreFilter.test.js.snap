// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from GenreFilter matches snapshot 1`] = `
<div>
  <div
    class="filter"
  >
    <p
      class="filterText"
    >
      Filter:
    </p>
    <select
      class="selection"
      id="genres"
      name="genres"
    >
      <option
        disabled=""
        value=""
      >
        Genres...
      </option>
      <option
        value="ALL"
      >
        ALL
      </option>
      <option
        value="Racing"
      >
        Racing
      </option>
      <option
        value="Fighting"
      >
        Fighting
      </option>
      <option
        value="Shooter"
      >
        Shooter
      </option>
      <option
        value="Puzzle"
      >
        Puzzle
      </option>
      <option
        value="Role-playing (RPG)"
      >
        Role-playing (RPG)
      </option>
      <option
        value="Simulator"
      >
        Simulator
      </option>
      <option
        value="Platform"
      >
        Platform
      </option>
      <option
        value="Adventure"
      >
        Adventure
      </option>
      <option
        value="Quiz/Trivia"
      >
        Quiz/Trivia
      </option>
      <option
        value="Sport"
      >
        Sport
      </option>
    </select>
  </div>
</div>
`;
